pkg/controllers/cspc-controller/sync.go

```diff
@@ -261,3 +270,63 @@ func (c *Controller) GetCSPIWithoutDeployment(cspc *cstor.CStorPoolCluster) ([]c
 	}
 	return cspiList, nil
 }
+
+// syncCSPI propagates all the required changes from cspc to respective cspi.
+func (pc *PoolConfig) syncCSPI(cspc *cstor.CStorPoolCluster) error {
+	cspiList, err := pc.Controller.GetCSPIListForCSPC(cspc)
+	if err != nil {
+		return errors.Wrapf(err, "failed to sync cspi(s) from its parent cspc %s", cspc.Name)
+	}
+	if len(cspiList.Items) == 0 {
+		return errors.Wrapf(err, "No cspi(s) found while trying to sync cspi(s) from its parent cspc %s", cspc.Name)
+	}
+
+	for _, cspi := range cspiList.Items {
+		cspi := cspi
+		err := pc.syncCSPIWithCSPC(cspc, &cspi)
+		if err != nil {
+			klog.Errorf("failed to sync cspi %s from its parent cspc %s: {%s}", cspi.Name, cspc.Name,err)
+		}
+	}
+	return nil
+}
+
+func (pc *PoolConfig) syncCSPIWithCSPC(cspc *cstor.CStorPoolCluster, cspi *cstor.CStorPoolInstance) error {
+	cspiCopy := cspi.DeepCopy()
+	klog.V(2).Infof("Syncing cspi %s from parent cspc %s", cspiCopy.Name, cspc.Name)
+	for _, poolSpec := range cspc.Spec.Pools {
+		if reflect.DeepEqual(poolSpec.NodeSelector,cspi.Spec.NodeSelector){
```

Comment:
By seeing this check looks like we shouldn't allow updating or modifying node selector(Just gathering Info).